### PR TITLE
Remove unnecessary group name validation in provisioning_api

### LINKED ISSUE #20817: UTF-8 support different on User Provisioning API function calls
I'm trying to do some operations on an owncloud 8.2 instance by using the User Provisioning API.
I've written a small bash script which basically hides the API and and provides a very narrow and simple interface. Within this script I'm using curl to communicate with the API.
I'm running into problems when I'm trying to add a group which has german umlauts in its name (e.g. "Däst"). I'm getting a statuscode 101/message 'Invalid group name'. 
Adding the same group name by using the OC webinterface succeeds.

The strange thing is that <i>I can delete the group by using my script without any problems!</i>

This is the curl call to add a group (no success):
<code>curl --netrc-file $credentials_file -X POST -k $base/groups -d groupid="$1" </code>
I also tried the variation "groupid=$1", no success
  
This call deletes the group successfully:
<code>curl --netrc-file $credentials_file -X DELETE -k $base/groups/"$1"</code>

my locale settings:
<code>LANG=de_DE.UTF-8</code>
<code>LC_CTYPE="de_DE.UTF-8"</code>
<code>LC_NUMERIC=de_DE.UTF-8</code>
<code>LC_TIME=de_DE.UTF-8</code>
<code>LC_COLLATE=de_DE.UTF-8</code>
<code>LC_MONETARY=de_DE.UTF-8</code>
<code>LC_MESSAGES="de_DE.UTF-8"</code>
<code>LC_PAPER="de_DE.UTF-8"</code>
<code>LC_NAME="de_DE.UTF-8"</code>
<code>LC_ADDRESS="de_DE.UTF-8"</code>
<code>LC_TELEPHONE="de_DE.UTF-8"</code>
<code>LC_MEASUREMENT=de_DE.UTF-8</code>
<code>LC_IDENTIFICATION="de_DE.UTF-8"</code>
<code>LC_ALL=</code>

The owncloud instance lives on a OpenSuse 13.2 machine
php 5.6, Apache 2.4.10, mysql 5.6.26

I have not startet a systematic test of all API calls yet, so this behaviour might be shown by
other API calls too.

Regards
Juergen

### PR DESCRIPTION
**Pull request by tomneedham**
Fixes https://github.com/owncloud/core/issues/20817 by removing the unnecessary group name validation that is not present elsewhere in ownCloud - allows the full range of characters.

WIP as I need to fix my env so I can check the tests.

### CODE REVIEW

File: apps/provisioning_api/tests/groupstest.php [hunk 1]
```diff
@@ -365,6 +365,27 @@ public function testAddGroup() {
 		$this->assertTrue($result->succeeded());
 	}
 
+	public function testAddGroupWithSpecialChar() {
+		$this->request
+			->method('getParam')
+			->with('groupid')
+			->willReturn('Iñtërnâtiônàlizætiøn');
```
Comment (MorrisJobke) [inline apps/provisioning_api/tests/groupstest.php]: Lame stuff compared to https://github.com/owncloud/core/pull/10935 :stuck_out_tongue_winking_eye:

Comment (tomneedham) [inline apps/provisioning_api/tests/groupstest.php]: Now that's what I was looking for!

Comment (rullzer) [inline apps/provisioning_api/tests/groupstest.php]: I guess this is simple enough. We just need to make sure proper tests are run on the group manager.

**Comment by rullzer** [comment]
Rebased and fixed the unit test (trivial).
Good to go from my POV

CC: @PVince81 @MorrisJobke @nickvergessen @LukasReschke

File: apps/provisioning_api/lib/groups.php [hunk 2]
```diff
@@ -130,7 +130,7 @@ public function getGroup($parameters) {
 	public function addGroup($parameters) {
 		// Validate name
 		$groupId = $this->request->getParam('groupid', '');
-		if( preg_match( '/[^a-zA-Z0-9 _\.@\-]/', $groupId ) || empty($groupId)){
+		if(empty($groupId)){
 			\OCP\Util::writeLog('provisioning_api', 'Attempt made to create group using invalid characters.', \OCP\Util::ERROR);
```
Comment (PVince81) [inline apps/provisioning_api/lib/groups.php]: update the error message ?

**Comment by MorrisJobke** [comment]
:+1:

**Comment by MorrisJobke** [comment]
I rebased on a more recent commit  to trigger the tests again ;9

**Comment by SergioBertolinSG** [comment]
:+1: Works fine.

Just a question, no character has to be forbidden in group names?

**Comment by rullzer** [comment]
@SergioBertolinSG that part should be handled in the GroupManager... so if chars are forbidden they should be forbidding in all code paths.

**Comment by SergioBertolinSG** [comment]
This needs to be backported to stable8.2.

**Comment by PVince81** [comment]
"needs" as in: for API consistency but also to make integration tests pass there

**Pull request merged**